zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineLoader.kt

```diff
@@ -186,9 +186,10 @@ class ZiplineLoader internal constructor(
           try {
             val networkZipline = loadFromManifest(applicationName, networkManifest, initializer)
             cachingFetcher?.pin(applicationName, networkManifest) // Pin after successful loads.
-            emit(LoadedZipline(networkZipline, networkManifest.freshAtEpochMs))
+            emit(LoadAttempt(LoadStatus.SUCCESS, networkZipline, networkManifest.freshAtEpochMs))
             previousManifest = networkManifest.manifest
           } catch (e: Exception) {
+            emit(LoadAttempt(LoadStatus.FAILURE, null, networkManifest.freshAtEpochMs))
             cachingFetcher?.unpin(applicationName, networkManifest) // Unpin after failed loads.
             throw e
```

Comment:
@swankjesse should we still be throwing here if we're now emitting on failure?

Reply:
It's certainly not an amazing pattern we got forced into here. Generally you want to handle or rethrow and never both. Maybe the design of `withLifecycleEvents` needs to change a bit in a future change to try and remedy this.